crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -130,18 +130,59 @@ def check_all_repos
       end
 
       def provided_and_enabled?(feature, platform = nil, arch = nil)
+        provided_with_enabled?(feature, platform, arch, true)
+      end
+
+      def provided?(feature, platform = nil, arch = nil)
+        provided_with_enabled?(feature, platform, arch, false)
+      end
+
+      def platform_available?(platform, arch)
+        available = true
+
+        repositories(platform, arch).each do |repo|
+          r = new(platform, arch, repo)
+
+          next if r.required != "mandatory"
+          next if r.active?
+
+          available = false
+          break
+        end
+
+        available
+      end
+
+      def disabled_platforms(arch)
+        # forcefully reload the data, as we don't want to have outdated info
+        # about what is mandatory or not
+        load!
+        all_platforms.reject { |platform| platform_available?(platform, arch) }
+      end
+
+      def admin_ip
+        @admin_ip ||= NodeObject.admin_node.ip
+      end
+
+      def web_port
+        @web_port ||= Proposal.where(barclamp: "provisioner").first.raw_attributes["web_port"]
+      end
+
+      private
+
+      def provided_with_enabled?(feature, platform = nil, arch = nil, check_enabled = true)
         answer = false
 
         if platform.nil?
           all_platforms.each do |p|
-            if provided_and_enabled?(feature, p)
+            if provided_and_enabled?(feature, p, arch, check_enabled)
```

Comment:
Maybe I am overlooking something here, but this calls the provided_and_enabled? method which does nothing but calling this method  (provided_with_enabled?) again, and also has the wrong number of arguments?